specs/menu/menu.md

```diff
@@ -0,0 +1,213 @@
+
+```html
+<fast-menu>
+    <fast-menu-item>Menu item one</fast-menu-item>
+    <fast-menu-item>Menu item two</fast-menu-item>
+    <fast-menu-item>
+        Menu item three
+        <fast-menu>
+            <fast-menu-item slot="item">Nested item one</fast-menu-item>
+            <fast-menu-item slot="item">Nested item two</fast-menu-item>
+            <fast-menu-item>
+                Nested item three
+                <fast-menu>
+                    <fast-menu-item slot="item">Nested item one</fast-menu-item>
+                    <fast-menu-item slot="item">Nested item two</fast-menu-item>
+                </fast-menu>
+            </fast-menu-item>
+        </fast-menu>
+    </fast-menu-item>
+    <fast-menu-item>Menu item four</fast-tree-item>
+</fast-context-menu>
+```
+
+#### Icon support:
+
+Consider a scenario where we have the following menu - some menu items have glyphs, others do not:
+```html
+<fast-menu>
+    <fast-menu-item slot="item">Menu item one</fast-menu-item>
+    <fast-menu-item slot="item">Menu item two</fast-menu-item>
+    <fast-menu-item slot="item">
+        <fancy-icon slot="icon"></fany-icon>
+        Menu item three
+    </fast-tree-item>
+</fast-context-menu>
+```
+The third item in the menu has a glyph, which would inset the text from the other menu items. To ensure that the content of this menu item aligns with the rest of the items, all items need to visually shift to ensure alignmnet of the menu content. This presents an interesting challenge for how we will communicate these changes to ensure the styling is appropriately applied. Ultimately, the parent just needs to know that *one* of the items below has a menu item with the glyph.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/context-menu)
+- [Material UI](https://material-ui.com/components/menus/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/menus/)
+- [Ant Design - menu](https://ant.design/components/menu/)
+- [Ant Design - dropdown](https://ant.design/components/menu/)
+- [Atlassian - menu](https://atlaskit.atlassian.com/packages/core/menu)
+- [Atlassian - dropdown menu](https://atlaskit.atlassian.com/packages/core/dropdown-menu)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/tree-view)
+
+---
+
+## Design
+
+### API
+
+*The key elements of the component's public API surface:*
+
+**Menu**
+*Component name:*
+- `fast-menu`
+
+*Attributes:*
+- extends HTML Element attributes
+
+*Slots:*
+- default slot for items
+
+*Events*
+- none
+
+**Menu Item**
+*Component name:*
+- `fast-menu-item`
+
+*Attributes:*
+- role - an enum representing the menu items' role
+    - menuitem
+    - menuitemcheckbox
+    - menuitemradio
+- disabled - the menu item is disabled
+- checked - sets the checked value for menuitemcheckbox or menuitemradio items
+
+*Slots:*
+- content - slot for the content (the default slot for the item)
+- item - the optional children
```

Comment:
I'm a little confused about this, it appears to be how you're determining nesting but it seems unintuitive.  Can you go into a little more detail?

Reply:
Can you elaborate on what is unintuitive about using a slot for creating a nested menu? I'm following a previous implementation and precedent with tree items where the default slot is for the content of the menu item and slots would be nested items.

Reply:
I see how this was a bit confusing.

Reply:
What is the expectation? Does it expect a menu element w/ items?

Reply:
I think so, I don't think I'd expect individual items without a menu wrapping them.

Reply:
nevermind - your examples below answered that question. I think "submenu" would work, though we could also align to "menu" which might provide some consistency when we create toolbar components that create slots for *top level* menus